USBPcapDriver/USBPcapTables.c

```diff
@@ -42,20 +42,25 @@ VOID USBPcapAddEndpointInfo(IN PRTL_GENERIC_TABLE table,
 {
     USBPCAP_INTERNAL_ENDPOINT_INFO  info;
     BOOLEAN                         new;
+    PUSBPCAP_INTERNAL_ENDPOINT_INFO pInfo;
 
     info.info.handle          = pipeInfo->PipeHandle;
     info.info.type            = pipeInfo->PipeType;
     info.info.endpointAddress = pipeInfo->EndpointAddress;
     info.info.deviceAddress   = deviceAddress;
 
-    RtlInsertElementGenericTable(table,
+    pInfo = RtlInsertElementGenericTable(table,
                                  (PVOID)&info,
                                  sizeof(USBPCAP_INTERNAL_ENDPOINT_INFO),
                                  &new);
 
     if (new == FALSE)
     {
-        DkDbgStr("Element already exists in table");
+        DkDbgStr("Element already exists in table. Lets replace elements data!");
+        pInfo->info.type            = pipeInfo->PipeType;
```

Comment:
This can lead to Blue Screen of Death, especially when driver verifier low resources simulation is active.

`If no matching element is found, but the new element cannot be inserted (for example, because the AllocateRoutine fails), RtlInsertElementGenericTable returns NULL.`

I think the easiest way to prevent it, is to change the if condition to `((new == FALSE) && (pInfo != NULL)`.

Reply:
Hm... might be true. I will correct it, in your way.